servicebroker/lib/routes/binding.js

```diff
@@ -70,53 +71,54 @@ module.exports = function(app, settings, catalog) {
-              }
-
-              rollbackTransaction(t, res, 500, {});
-            });
-          } else if (length > 1) { // an app has been bound to more than one service instance, this error should not exist
-            logger.error("Bind failed: duplicate bind", { app_guid: appId });
-            res.status(409).json({ "description": messageUtil.getMessage("DUPLICATE_BIND", { "applicationId": appId }) });
-            return;
-          } else if (length == 1) { // an app has been bound to a service instance
-            var bindingRecord = result[0];
-            if (bindingRecord.serviceInstanceId === serviceInstanceId) {
-              logger.error("Bind failed: app already bound", { app_guid: appId, serviceInstanceId: serviceInstanceId });
-              res.status(409).json({});
+            }
+
+          }).catch(function(error1) { //catch findorcreate
+            logger.error("Bind failed: create error", { error: error1 });
+            if (error1 instanceof models.sequelize.UniqueConstraintError) {
+              rollbackTransaction(t, res, 409, {});
+              return;
+            } else if (error1 instanceof models.sequelize.ForeignKeyConstraintError) {
+              rollbackTransaction(t, res, 404, { "description": messageUtil.getMessage("SERVICEINSTANCE_NOT_EXIST", { "serviceInstanceId": serviceInstanceId }) });
               return;
             }
 
-            logger.error("Bind failed: duplicate bind", { app_guid: appId });
-            res.status(409).json({ "description": messageUtil.getMessage("DUPLICATE_BIND", { "applicationId": appId }) });
+            rollbackTransaction(t, res, 500, {});
+          });
+        } else if (length > 1) { // an app has been bound to more than one service instance, this error should not exist
+          logger.error("Bind failed: duplicate bind", { app_guid: appId });
+          res.status(409).json({ "description": messageUtil.getMessage("DUPLICATE_BIND", { "applicationId": appId }) });
+          return;
+        } else if (length == 1) { // an app has been bound to a service instance
+          var bindingRecord = result[0];
+          if (bindingRecord.serviceInstanceId === serviceInstanceId) {
+            logger.error("Bind failed: app already bound", { app_guid: appId, serviceInstanceId: serviceInstanceId });
+            res.status(409).json({});
+            return;
           }
-        });
-      }).catch(function(error2) { //catch transaction
-        logger.error("Bind failed: transaction error", { error: error2 });
-        rollbackTransaction(t, res, 500, {});
+
+          logger.error("Bind failed: duplicate bind", { app_guid: appId });
+          res.status(409).json({ "description": messageUtil.getMessage("DUPLICATE_BIND", { "applicationId": appId }) });
+        }
       });
-    }
+    }).catch(function(error2) { //catch transaction
+      logger.error("Bind failed: transaction error", { error: error2 });
+      rollbackTransaction(t, res, 500, {});
```

Comment:
@pradyutsarma Just went through sequilze transaction doc(http://docs.sequelizejs.com/manual/tutorial/transactions.html). The pattern is 
sequelize.transaction().then(function (t) {});
so remove the catch part here.